src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -772,16 +778,7 @@ def prepare_inputs_for_generation(self, input_ids, past=None, **kwargs):
             input_ids = input_ids[:, -1].unsqueeze(-1)
 
         attention_mask = kwargs.get("attention_mask", None)
-        position_ids = kwargs.get("position_ids", None)
-
-        if attention_mask is not None and position_ids is None:
-            # create position_ids on the fly for batch generation
-            position_ids = attention_mask.long().cumsum(-1) - 1
-            position_ids.masked_fill_(attention_mask == 0, 1)
-            if past:
-                position_ids = position_ids[:, -1].unsqueeze(-1)
-        else:
-            position_ids = None
```

Comment:
Same here

Reply:
Think the change is fine here. Maybe to clarify I think having added `position_ids` was simply a bug and is not a design change now. If I understand correctly BLOOM **never** made use of `position_ids` and always relies on `alibi`, so this change is ok for me

Reply:
Or was there a use case where BLOOM could make use of `position_ids` @thomasw21 ?

Reply:
Nope none, I mean none of the pretrained models make use of it during training. I can't think of anything where we might use it as of now (maybe if we introduce a recurrence a la transformer-XL that I've been meaning to experiment with, but I don't think I would rely on this argument and would probably introduce something else)